airflow/contrib/hooks/aws_sns_hook.py

```diff
@@ -53,8 +55,16 @@ def publish_to_target(self, target_arn, message):
             'default': message
         }
 
+        if subject is None:
```

Comment:
yes. it will work. To handle already existing code. I added this None check. if someone is not sending subject while calling method it will work.

Reply:
```
    def publish_to_target(self, target_arn, message, subject=None):
        """
        Publish a message to a topic or an endpoint.
        :param target_arn: either a TopicArn or an EndpointArn
        :type target_arn: str
        :param message: the default message you want to send
        :param message: str
        :param subject: subject of message
        :type subject: str
        """

        conn = self.get_conn()

        messages = {
            'default': message
        }

        return conn.publish(
            TargetArn=target_arn,
            Message=json.dumps(messages),
            MessageStructure='json',
            Subject=subject
        )
```
Will this code work properly?

Reply:
yes @mik-laj. Am I missing anything?

Reply:
So we don't need one additional "If" instruction in the code?

Reply:
let me test one more time @mik-laj

Reply:
@mik-laj  we need if condition because. boto3 is throwing an error when you call publish_to_target method with subject as None. 

Error message from boto3:
```
botocore.exceptions.ParamValidationError: Parameter validation failed:
```

Because boto3 expects the subject to be a string. not None.  To handle this case I added if condition. 

Example: 
1. when you want to send SNS notification with the subject 
```
AwsSnsHook().publish_to_target(target_arn='arn', message='test', subject='subject')
```
2. when you want to send SNS notification witout subject (AWS will allocate default subject)
```
AwsSnsHook().publish_to_target(target_arn='arn', message='test')
```

Let me know if you have any thoughts!

Reply:
Well, then this change looks good. I accept it